pkg/policies/engine/rego/rego.go

```diff
@@ -0,0 +1,90 @@
+//
+// Copyright 2024 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rego
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/chainloop-dev/chainloop/pkg/policies/engine"
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/rego"
+)
+
+// Rego policy checker for chainloop attestations and materials
+type Rego struct {
+}
+
+// Force interface
+var _ engine.PolicyEngine = (*Rego)(nil)
+
+func (r *Rego) Verify(ctx context.Context, policy *engine.Policy, input []byte) ([]*engine.PolicyViolation, error) {
+	policyString := string(policy.Module)
+	parsedModule, err := ast.ParseModule(policy.Name, policyString)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse rego policy: %w", err)
+	}
+
+	// Decode input as json
+	decoder := json.NewDecoder(bytes.NewReader(input))
+	decoder.UseNumber()
+	var decodedInput interface{}
+	if err := decoder.Decode(&decodedInput); err != nil {
+		return nil, fmt.Errorf("failed to parse input: %w", err)
+	}
+
+	// add input
+	regoInput := rego.Input(decodedInput)
+
+	// add module
+	regoFunc := rego.ParsedModule(parsedModule)
+
+	// add query. Note that the predefined rule to look for is `deny`
```

Comment:
Shall we document this?

Reply:
Yes, I plan to add documentation once we release this feature.